helpers/QtiPackageExporter.php

```diff
@@ -20,40 +20,99 @@
 
-use League\Flysystem\Util;
-use oat\oatbox\service\ConfigurableService;
+use common_Exception;
+use common_exception_Error;
+use common_report_Report;
+use oat\oatbox\filesystem\File;
+use oat\oatbox\filesystem\FileSystemService;
+use oat\tao\helpers\FileHelperService;
+use oat\tao\model\service\InjectionAwareService;
 use taoQtiTest_models_classes_export_TestExport22 as TestExporter;
-use core_kernel_classes_Resource as RdfResource;
-use tao_helpers_File as FileHelper;
 
-class QtiPackageExporter extends ConfigurableService
+class QtiPackageExporter extends InjectionAwareService
 {
+    public const SERVICE_ID = 'taoQtiTest/QtiPackageExporter';
+    public const QTI_PACKAGE_FILENAME = 'qti_test_export';
+
+    /** @var TestExporter */
+    private $testExporter;
+
+    /** @var FileSystemService */
+    private $fileSystemService;
+
+    /** @var FileHelperService */
+    private $fileHelperService;
+
+    public function __construct(
+        TestExporter $testExporter,
+        FileSystemService $fileSystemService,
+        FileHelperService $fileHelperService
+    ) {
+        $this->testExporter = $testExporter;
+        $this->fileSystemService = $fileSystemService;
+        $this->fileHelperService = $fileHelperService;
+    }
+
     /**
-     * @param RdfResource $test
+     * @param string $testUri
      *
      * @return array
-     * @throws \common_Exception
-     * @throws \common_exception_Error
+     * @throws common_Exception
+     * @throws common_exception_Error
      */
-    public function exportDeliveryQtiPackage(RdfResource $test)
+    public function exportDeliveryQtiPackage(string $testUri)
     {
-        $exportReport = $this->getTestExporter()->export(
+        $exportReport = $this->testExporter->export(
             [
-            'filename' => Util::normalizePath('qti_package_'),
-            'instances' => $test->getUri(),
-            'uri' => $test->getUri()
+                'filename' => self::QTI_PACKAGE_FILENAME,
+                'instances' => $testUri,
+                'uri' => $testUri
             ],
-            FileHelper::createTempDir()
+            $this->fileHelperService->createTempDir()
         );
 
+        if ($exportReport->getType() === common_report_Report::TYPE_ERROR) {
+            throw new common_Exception('QTI Test package export failed.');
+        }
+
         return $exportReport->getData();
     }
 
     /**
-     * @return TestExporter
+     * @param string $testUri
+     * @param string $fileSystemId
+     * @param string $filePath
+     * @return File
+     * @throws common_Exception
+     */
+    public function exportQtiTestPackageToFile(string $testUri, string $fileSystemId, string $filePath): File
+    {
+        $result = $this->exportDeliveryQtiPackage($testUri);
+        if (!isset($result['path']) || !is_string($result['path'])) {
+            throw new common_Exception('Export report does not contain path to exported QTI package: ' . json_encode($result));
+        }
```

Comment:
Maybe I missed that and it's done in another level, but shouldn't we also verify if path exist?